src/main/java/com/palantir/gradle/graal/GradleGraalPlugin.java

```diff
@@ -45,60 +43,39 @@
 public class GradleGraalPlugin implements Plugin<Project> {
 
     /** Location to cache downloaded and extracted GraalVM tooling. */
-    public static final Path CACHE_DIR =
-            Paths.get(System.getProperty("user.home"), ".gradle", "caches", "com.palantir.graal");
+    static final Path CACHE_DIR = Optional.ofNullable(System.getProperty("com.palantir.graal.cache.dir"))
```

Comment:
I needed to add an override here otherwise all the tests would write to your actual home dir and thereby not be reproducible!

Reply:
This can carry over if running tests without forking each time.
Ideally this'd be not static and a read from a gradle project property (`gradle.properties` / `-P` style).